csvutil/csv.go

```diff
@@ -27,6 +27,34 @@ func min(a, b int) int {
 	return b
 }
 
+func formatValue(val interface{}) (string, error) {
+	var err error
+	rv := reflect.Indirect(reflect.ValueOf(val))
+	rt := rv.Type()
+	switch rt.Kind() {
+	case reflect.Bool:
+		return strconv.FormatBool(rv.Bool()), nil
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		return strconv.FormatInt(rv.Int(), 10), nil
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		return strconv.FormatUint(rv.Uint(), 10), nil
+	case reflect.Float32, reflect.Float64:
+		return strconv.FormatFloat(rv.Float(), 'g', -1, rt.Bits()), nil
+	case reflect.String:
+		return rv.String(), nil
+	case reflect.Slice:
+		vals := make([]string, rv.Len())
+		for i := 0; i < rv.Len(); i++ {
+			if vals[i], err = formatValue(rv.Index(i).Interface()); err != nil {
+				return "", err
+			}
+		}
+		return "[" + strings.Join(vals, ",") + "]", nil
```

Comment:
is this a well known or widespread way to deal with non-scalar values in the CSV world?
what happens if the `csv.Writer.Comma` is `','` instead? wouldn't this confuse other tools?

Reply:
You're right, it would cause confusion in that case. What pandas seems to do when writing non scalars is surrounding the array with `"` marks. That should resolve any ambiguity, does that sound fine?

Reply:
makes sense.
what does it do for:
- slices of slices of ints?
- slices of strings?

(so it's a pandas-only thing? or is it more widespread?)

Reply:
ok:

Reply:
ok:

```
>>> df = pd.DataFrame({'slice': [[[10,11],[2,3]],[[40,41],[5,6]]]})
>>> df.to_csv("foo.csv")
```

leads to:
```
,slice
0,"[[10, 11], [2, 3]]"
1,"[[40, 41], [5, 6]]"
```

and:
```
>>> df = pd.DataFrame({'slice': [[["10","11"],["2","3"]],[["40","41"],["5","6"]]]})
>>> df.to_csv("foo.csv")
```
to:
```
,slice
0,"[['10', '11'], ['2', '3']]"
1,"[['40', '41'], ['5', '6']]"
```

so if we "standardize" on what pandas does, we should try to replicate that behaviour (with a unit test :))
```
```

Reply:
Oh, it looks like Go's csv automatically adds quotes if a comma is present in the field: https://golang.org/src/encoding/csv/writer.go?s=1503:1548#L38  
This was an interesting was to discover it :)